staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4137,6 +4137,11 @@ type ServiceSpec struct {
 	// If this is not specified or empty, no topology constraints will be applied.
 	// +optional
 	TopologyKeys []string `json:"topologyKeys,omitempty" protobuf:"bytes,16,opt,name=topologyKeys"`
+
+	// allocateLoadBalancerNodePorts defines if NodePorts shall be automatically for services with type LoadBalancer.
+	// Default is "true". It may be set to "false" in clusters that don't have an external load-balancer that uses NodePorts.
+	// This is often the case in private clouds that for instance uses "metallb".
```

Comment:
This needs to match the comment from the internal type

Reply:
+1

Also - mark it as +optional